zerver/lib/widget.py

```diff
@@ -30,8 +30,11 @@ def do_widget_post_save_actions(message: MutableMapping[str, Any]) -> None:
 
     widget_type = None
     extra_data = None
-    if content in ['/poll', '/tictactoe']:
-        widget_type = content[1:]
+
+    tokens = content.split(' ')
```

Comment:
I think you need to handle that `tokens[0]` could theoretically throw `IndexError` or whatever.

I would consider something like this. 

~~~ py
def get_widget_type() -> Optional[str]:
    valid_widget_types = ['tictactoe', 'poll']
    tokens = content.split(' ')
    if not tokens:
        return None

    if tokens[0].startswith('/'):
        widget_type = tokens[0][1:]
        if widget_type in valid_widget_types:
            return widget_type
    return None

widget_type = get_widget_type()
if widget_type is None:
    return